vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java

```diff
@@ -240,13 +284,31 @@ protected void write(FileArchive fa, boolean notifyModuleManager)
     }
   }
 
+  /**
+   * Saves the archive, always prompting for a new filename. If a defaultExtension has been
+   * provided, it will be added to the filename unless the user specifies a different one explicitly.
+   * @param notifyModuleManager If true, notifies Module Manager that the save has occurred
+   * @throws IOException IOException
+   */
   public void saveAs(boolean notifyModuleManager) throws IOException {
     final FileChooser fc = FileChooser.createFileChooser(
       GameModule.getGameModule().getPlayerWindow(),
       (DirectoryConfigurer) Prefs.getGlobalPrefs()
                                  .getOption(Prefs.MODULES_DIR_KEY));
     if (fc.showSaveDialog() != FileChooser.APPROVE_OPTION) return;
-    final String filename = fc.getSelectedFile().getPath();
+    String filename = fc.getSelectedFile().getPath();
+
+    if (!StringUtils.isEmpty(defaultExtension) && (filename.lastIndexOf('.') < 0)) {
+      if (defaultExtension.indexOf('.') >= 0) {
+        filename = filename + defaultExtension;
+      }
+      else {
+        filename = filename + "." + defaultExtension;
+      }
```

Comment:
Would it be simpler to expect the default extension to supply its own `.`, and then not check for it here?

Reply:
Fair enough. I'm probably overengineering given the small number of places/times this needs to get called.